qemu/qmp_test.go

```diff
@@ -810,3 +810,42 @@ func TestQMPPCIDeviceAdd(t *testing.T) {
 	q.Shutdown()
 	<-disconnectedCh
 }
+
+// Checks that CPU are correctly added using device_add
+func TestQMPCPUDeviceAdd(t *testing.T) {
+	connectedCh := make(chan *QMPVersion)
+	disconnectedCh := make(chan struct{})
+	buf := newQMPTestCommandBuffer(t)
+	buf.AddCommand("device_add", nil, "return", nil)
+	cfg := QMPConfig{Logger: qmpTestLogger{}}
+	q := startQMPLoop(buf, cfg, connectedCh, disconnectedCh)
+	checkVersion(t, connectedCh)
+	driver := "qemu64-x86_64-cpu"
+	cpuID := "cpu-0"
+	socketID := "0"
+	coreID := "1"
+	threadID := "0"
+	err := q.ExecuteCPUDeviceAdd(context.Background(), driver, cpuID, socketID, coreID, threadID)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	q.Shutdown()
+	<-disconnectedCh
+}
+
+// Checks that hotpluggable CPUs are listed correctly
+func TestQMPExecuteQueryHotpluggableCPUs(t *testing.T) {
+	connectedCh := make(chan *QMPVersion)
+	disconnectedCh := make(chan struct{})
+	buf := newQMPTestCommandBuffer(t)
+	buf.AddCommand("query-hotpluggable-cpus", nil, "return", nil)
```

Comment:
This isn't actually testing the function as it doesn't pass in any response data.  We should pass valid information for the last parameter of AddCommand and check it is correctly converted into a slice of HotpluggableCPU CPUs.

Reply:
this is an interesting issue, because ```AddCommand``` receives a ```map``` and what I need is a ```slice```, I'll add a new test result structure (```qmpTestResultSlice```)  and a new member of this type to ```qmpTestCommandBuffer```, then I'll implement a new version of ```AddCommand```, otherwise I'd have to modify ```qmpTestResult``` to support slices and patch all the ```qmp_test.go``` file

anyway lemme check what I can do

Reply:
I think AddCommand should be modified to take an interface{} instead of a map.  This would make more sense.  Note this parameter has never been used before which is why it was defined incorrectly.  As none of the other tests are currently using it, changing the data parameter to interface{} won't break anything. You'd also need to change qmpTestResult.data